x/auth/ante/sigverify.go

```diff
@@ -250,6 +261,46 @@ func OnlyLegacyAminoSigners(sigData signing.SignatureData) bool {
 	}
 }
 
+func verifyIsOnCurve(pubKey cryptotypes.PubKey) (err error) {
+	switch typedPubKey := pubKey.(type) {
+	case *ed25519.PubKey:
+		return errorsmod.Wrap(sdkerrors.ErrInvalidPubKey, "ED25519 public keys are unsupported")
+	case *secp256k1.PubKey:
+		pubKeyObject, err := secp256k1dcrd.ParsePubKey(typedPubKey.Bytes())
+		if err != nil {
+			if strings.Contains(err.Error(), "not on the secp256k1 curve") {
```

Comment:
Can we use `errors.Is` if possible? Check if `secp256k1dcrd` package has a typed public error type we can check against.